### PR TITLE
scikit-optimize: 0.6 -> 0.7.4

### PR DESCRIPTION
**Pull request by jethrokuan**
###### Motivation for this change
Updates scikit-optimize to 0.7.4.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/scikit-optimize/default.nix [hunk 1]
```diff
@@ -10,16 +11,17 @@
 
 buildPythonPackage rec {
   pname = "scikit-optimize";
-  version = "0.6";
+  version = "0.7.4";
```
Comment (jonringer) [inline pkgs/development/python-modules/scikit-optimize/default.nix:14]:
```suggestion
  version = "0.7.4";
  disabled = isPy27;
```

```
builder for '/nix/store/hn3ld9bfc7sh4yrnfgwdq2gqq8x5sxxv-python2.7-scikit-optimize-0.7.4.drv' failed with exit code 1; last 10 log lines:
  removing build/bdist.linux-x86_64/wheel
  Finished executing setuptoolsBuildPhase
  installing
  Executing pipInstallPhase
  /build/source/dist /build/source
  DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
  Processing ./scikit_optimize-0.7.4-py2.py3-none-any.whl
  Requirement already satisfied: pyaml>=16.9 in /nix/store/akhmhpgra73q362577j7vc69qqfhvhn6-python2.7-pyaml-19.12.0/lib/python2.7/site-packages (from scikit-optimize==0.7.4) (19.12.0)
  ERROR: Could not find a version that satisfies the requirement joblib>=0.11 (from scikit-optimize==0.7.4) (from versions: none)
  ERROR: No matching distribution found for joblib>=0.11 (from scikit-optimize==0.7.4)
```

File: pkgs/development/python-modules/scikit-optimize/default.nix [hunk 2]
```diff
@@ -1,6 +1,7 @@
 { lib
 , buildPythonPackage
 , fetchFromGitHub
```
Comment (jonringer) [inline pkgs/development/python-modules/scikit-optimize/default.nix:3]:
```suggestion
, fetchFromGitHub
, isPy27
```

**Review by jonringer — COMMENTED** [review]
please ensure the package still builds

File: pkgs/development/python-modules/scikit-optimize/default.nix (hunk 2 above)
Comment (jethrokuan) [reply pkgs/development/python-modules/scikit-optimize/default.nix:3]: Sorry, was intending to get back to this, was attending to something else.

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, the commit message should be:
```
python3Packages.scikit-optimize: 0.6 -> 0.7.4
```

also, please squash all fixups

**Comment by risicle** [comment]
Superseded by #97990

**Pull request closed without merging**